compiler/luci/service/src/Nodes/CircleReshape.cpp

```diff
@@ -34,4 +62,107 @@ luci::CircleNode *CloneNodeLet<CN::OPQR>::visit(const luci::CircleReshape *node)
   return cloned;
 }
 
+namespace sinf
+{
+
+/**
+ * @note  CircleReshape has new shape info in two places: 2nd input and attribute.
+ *        This shape inference uses shape from input 'shape' node when it's constant.
+ *        If not, shape will be from node itself. shape from attribute is not used.
+ *
+ * TODO Change this policy when not appropriate
+ */
+loco::TensorShape Algorithm::visit(const luci::CircleReshape *node)
```

Comment:
Do you plan to open this PR as-is (as a first step - to migrate original algorithm to new place)? 

Or, Do you implement a dynamic shape inference algorithm as discussed here (https://github.com/Samsung/ONE/issues/13927#issuecomment-2328295368)?

Reply:
I'm planning to implement everything (migration, dynamic shape inference) in this draft PR.
After that, I was going to split this draft PR into migration PR and dynamic shape inference PR (or even smaller PRs).
But if you think the migration PR should be open ASAP, I will open this PR first as-is (with more test cases).

Reply:
> I'm planning to implement everything (migration, dynamic shape inference) in this draft PR 

:+1: 

> But if you think the migration PR should be open ASAP, I will open this PR first as-is (with more test cases). 

Either way is fine with me. I was just curious about your plans :)

Reply:
Ah, Also i guess you might have trouble because of `Net_Gather_SparseToDense_AddV2_000`. 

I think you could avoid this problem by adding `optimize(Net_Gather_SparseToDense_AddV2_000`) in `compiler/common-artifacts/exclude.lst`.

Reply:
The recipe modification has been completed (https://github.com/Samsung/ONE/issues/13939). You can proceed without excluding the problematic recipe.